src/openssh/patch/0002-Export-remote-info-for-authorization.patch

```diff
@@ -0,0 +1,89 @@
+From ea2574ea6fac2feeae506d7903a6164081e66d2b Mon Sep 17 00:00:00 2001
+From: liuh-80 <liuh@microsoft.com>
+Date: Fri, 30 Sep 2022 16:57:03 +0800
+Subject: [PATCH] Export remote info for authorization.
+ authorization.
+
+---
+ auth.c    | 13 +++++++++++++
+ auth.h    |  3 +++
+ session.c |  3 +++
+ sshd.c    |  5 +++++
+ 4 files changed, 24 insertions(+)
+
+diff --git a/auth.c b/auth.c
+index c3693ba3f..4123b78ee 100644
+--- a/auth.c
++++ b/auth.c
+@@ -914,3 +914,16 @@ auth_authorise_keyopts(struct ssh *ssh, struct passwd *pw,
+ 
+ 	return 0;
+ }
++
++/* Export remote IP address and port for authorization. */
++void
++export_remote_info(struct ssh *ssh)
++{
++	const char *remote_ip = ssh_remote_ipaddr(ssh);
++	setenv("SSH_CLIENT_IPADDR", remote_ip, 1);
++
++	const int remote_port = ssh_remote_port(ssh);
++	const char remote_port_str[32];
++	snprintf(remote_port_str, sizeof(remote_port_str), "%d", remote_port);
++	setenv("SSH_CLIENT_IPADDR_PORT", remote_port_str, 1);
```

Comment:
>SSH_CLIENT_IPADDR_PORT [](http://example.com/codeflow?start=10&length=22)

Sorry for misleading. I was proposing one single env var `SSH_CLIENT_IPADDR_PORT` to contain both ip and port, which is similar to existing `SSH_CLIENT` var.

Reply:
Fixed, because environment variable format changed, will update TACACS PR after this PR merged.